src/vtquery.cpp

```diff
@@ -9,24 +9,95 @@
 // #include <mapbox/geometry/geometry.hpp>
 // #include <mapbox/geometry/algorithms/closest_point.hpp>
 #include <vtzero/vector_tile.hpp>
+#include <vtzero/types.hpp>
 // #include <mapbox/variant.hpp>
 
 namespace VectorTileQuery {
 
+struct TileObject {
+    TileObject(std::uint32_t z0,
+        std::uint32_t x0,
+        std::uint32_t y0,
+        v8::Local<v8::Object> buffer)
+      : z(z0),
+        x(x0),
+        y(y0),
+        data(node::Buffer::Data(buffer),node::Buffer::Length(buffer)) {
+           buffer_ref.Reset(buffer.As<v8::Object>());
+        }
+
+    // explicitly use the destructor to clean up
+    // the persistent buffer ref by Reset()-ing
+    ~TileObject() {
+        buffer_ref.Reset();
+    }
+
+    // guarantee that objects are not being copied by deleting the
+    // copy and move definitions
+
+    // non-copyable
+    TileObject( TileObject const& ) = delete;
+    TileObject& operator=(TileObject const& ) = delete;
+
+    // non-movable
+    TileObject( TileObject && ) = delete;
+    TileObject& operator=(TileObject && ) = delete;
+
+    std::uint32_t z;
+    std::uint32_t x;
+    std::uint32_t y;
+    vtzero::data_view data;
+    Nan::Persistent<v8::Object> buffer_ref;
+};
+
+struct QueryData {
+    QueryData(std::uint32_t num_tiles) {
+        tiles.reserve(num_tiles);
+    }
+    ~QueryData() = default;
+
+    // guarantee that objects are not being copied by deleting the
+    // copy and move definitions
+
+    // non-copyable
+    QueryData( QueryData const& ) = delete;
+    QueryData& operator=(QueryData const& ) = delete;
+
+    // non-movable
+    QueryData( QueryData && ) = delete;
+    QueryData& operator=(QueryData && ) = delete;
+
+    // buffers object thing
+    std::vector<std::unique_ptr<TileObject>> tiles;
+
+    // lng/lat
+    double latitude = 0.0;
+    double longitude = 0.0;
+
+    // options
+    double radius = 0.0;
+    std::uint32_t results_length = 5;
+    std::vector<std::string> layers{};
+    std::string geometry{};
+};
+
 struct Worker : Nan::AsyncWorker {
     using Base = Nan::AsyncWorker;
 
-    Worker(Nan::Callback* callback)
-        : Base(callback), result_{"hello"} {}
+    Worker(std::unique_ptr<QueryData> query_data,
+           Nan::Callback* callback)
+        : Base(callback),
+          query_data_(std::move(query_data)) {}
 
     // The Execute() function is getting called when the worker starts to run.
     // - You only have access to member variables stored in this worker.
     // - You do not have access to Javascript v8 objects here.
     void Execute() override {
         try {
+            // Get the object from the unique_ptr
+            //QueryData const& data = *query_data_;
```

Comment:
@mapsam Is the `//` intentional here? Is the Worker executing from somewhere else?

Reply:
We're not doing any actual work with `data` yet, so wasn't able to compile with an unused variable!

Reply:
wrt to `wasn't able to compile with an unused variable!` I've added `-Wno-error=unused-variable` to the recommended list, which disables just this one warning from becoming an error. No problem commenting/avoiding this warning of course - which is best practice -
 but I've found it a little annoying to have it be a hard error while developing. See https://github.com/mapbox/cpp/issues/37#issuecomment-336200744

Reply:
Sweet, thanks @mapsam . Key word --> `yet` :)